toolkit/count_spacers.py

```diff
@@ -145,9 +145,19 @@ def _process_fastq_file(self, file, library_sequences):
 	# TODO rewrite with pandas
 	def _get_library_sequences(self):
 		# Open library sequences file and initiate dictionary of read counts for each guide
+		library_sequences = {}
 		with io.open(self.config.library_file, newline=None) as infile:
 			reader = csv.reader(infile)
-			return {rows[0]: 0 for rows in reader}
+			for rows in reader:
+				try:
+					key = rows[0].upper()
+					# Remove any non-ASCII characters
+					key = ''.join(char for char in key if ord(char) < 128)
+					library_sequences[key] = 0
+				except UnicodeDecodeError:
+					# Handle encoding issues
+					print("Encoding issue with sequence:", rows[0])
+		return library_sequences
```

Comment:
This looks good, but could removing non-ASCII characters cause any issues @jasonniehaus ?

Reply:
We expect the library sequences to be guide RNA sequences (i.e. a string of A,C,G,T; possible N or other degenerate bases), so removing non-ASCII characters is OK, but not necessarily complete. Our intention is not to catch all possible errors here. @saksh2311 Let's remove this section, as it's not needed for the example files provided in the repo.

Reply:
okay